spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/TraceAutoConfiguration.java

```diff
@@ -51,4 +55,19 @@ public DefaultTracer traceManager(Sampler sampler,
 									ApplicationEventPublisher publisher) {
 		return new DefaultTracer(sampler, random(), publisher);
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public CounterService noOpSpanCounterService() {
+		return new NoOpSpanCounterService();
```

Comment:
I don't see why we need this. We should just opt out of collecting metrics if there is no `CounterService` available.